analyses/methylation-probe-annotations/01-create-bed-files.R

```diff
@@ -0,0 +1,160 @@
+# where multiple UTRs have been predicted for a transcript
+exons_introns <- gencode_bed |> 
+  dplyr::filter(!type %in% c("five_prime_UTR", "three_prime_UTR"))
+plus_five_utrs <-gencode_bed |> 
+  dplyr::filter(type == "five_prime_UTR", strand == "+") |> 
+  dplyr::group_by(seqid, type, transcript, strand) |> 
+  dplyr::summarize(start = min(start), end = min(end)) |> 
+  dplyr::arrange(seqid, start, end) |> 
+  ungroup()
+minus_five_utrs <-gencode_bed |> 
+  dplyr::filter(type == "five_prime_UTR", strand == "-") |> 
+  dplyr::group_by(seqid, type, transcript, strand) |> 
+  dplyr::summarize(start = min(start), end = min(end)) |> 
+  dplyr::arrange(seqid, start, end) |> 
+  ungroup()
+plus_three_utrs <-gencode_bed |> 
+  dplyr::filter(type == "three_prime_UTR", strand == "+") |> 
+  dplyr::group_by(seqid, type, transcript, strand) |> 
+  dplyr::summarize(start = min(start), end = min(end)) |> 
+  dplyr::arrange(seqid, start, end) |> 
+  ungroup()
+minus_three_utrs <-gencode_bed |> 
+  dplyr::filter(type == "three_prime_UTR", strand == "-") |> 
+  dplyr::group_by(seqid, type, transcript, strand) |> 
+  dplyr::summarize(start = min(start), end = min(end)) |> 
+  dplyr::arrange(seqid, start, end) |> 
+  ungroup()
+gencode_bed <- dplyr::bind_rows(exons_introns, plus_five_utrs, minus_five_utrs,
+                                plus_three_utrs, minus_three_utrs) |> 
+  dplyr::select(seqid, start, end, type, strand, transcript) |> 
+  dplyr::arrange(seqid, start, end, transcript) |> 
+  dplyr::distinct() 
+
+# # extend 3'-UTR by 1kb
+# gencode_bed <- gencode_bed |>
+#   dplyr::mutate(start = case_when(type == "three_prime_UTR" &
+#                                     strand == "-" ~ as.integer(start - 1000),
+#                                   TRUE ~ as.integer(start)),
+#                 end = case_when(type == "three_prime_UTR" &
+#                                   strand == "+" ~ as.integer(end + 1000),
+#                                 TRUE ~ as.integer(end)))
+
+
+# add three_prime_UTR_extension - 1kb extension beyond the 3'-UTR
+three_prime_UTR_extensions <- gencode_bed |>
+  dplyr::filter(type == "three_prime_UTR") |>
+  dplyr::mutate(type = "three_prime_UTR_extension",
+                start = case_when(strand == "-" ~ as.integer(start - 1001), 
+                                  strand == "+" ~  as.integer(end + 1)),
+                end = case_when(strand == "-" ~ as.integer(start + 1000),
+                                strand == "+" ~ as.integer(end + 1001)))
+
+
+# add promoter region - 1kb extension beyond the 5'-UTR
+promoters <- gencode_bed |>
+  dplyr::filter(type == "five_prime_UTR") |>
+  dplyr::mutate(type = "promoter",
+                start = case_when(strand == "-" ~ as.integer(end + 1), 
+                                  strand == "+" ~  as.integer(start - 1001)),
+                end = case_when(strand == "-" ~ as.integer(end + 1001),
+                                strand == "+" ~ as.integer(start + 1000)))
```

Comment:
On the "+" strand, a promoter region consisting of 1000 bases would be from `start - 1001` to `start - 1`

`end = case_when(strand == "+" ~ as.integer(start - 1)`

Reply:
@afarrel  and @jharenza, the code and results are ok too. 
 
On the "+" strand, a promoter region consisting of 1000 bases would be from  `start - 1001` to `start + 1000` because the `start` value has already been mutated from the original `start` to `start - 1001`. Therefore,  only need to add 1000 bases to the new mutated `start` to get the `end` coordinate equivalent to the original (unmutated) `start - 1`.

Check results produced intermediate bed file when you run the module - `inputs/gencode.v39.primary_assembly.gene_features.annotation.bed`

Reply:
@afarrel can you comment

Reply:
@ewafula is right. I double checked the annotations in the bed file for genes on both strands and it looks good.